### DISCUSSION TITLE
Built-in Github cookie, managed by the cookie consent manager.

### DISCUSSION THREAD

**Discussion by white-gorilla** [squidfunk/mkdocs-material#4666] [Q&A] [answered]
Hello,

Sorry, that's probably a beginner's question.

I have activated GA4 on our website and the cookie consent manager.
Everything works perfectly. The two GA cookies are created well, or not, depending on his choice. The tracking in GA works fine.

My question is about the built-in Github cookie, which is enabled by default. I don't understand what it does (github cookies are created if GA4 cookies are not allowed, and replaced by those of GA4 if it is activated. In any case, I don't see any queries or anything sent if Github cookies are enabled).

Can you help me?

Thanks in advance.

Vincent.

**Comment by squidfunk** [comment, marked as answer]
The GitHub "cookie" is needed for consenting to request [repository statistics](https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/#repository) from GitHub (top right). There's not really a cookie set, as it's just Material for MkDocs performing two requests against GitHub's API. It was pointed out in #4329 that the application performed these request without the user's prior consent, so when you enable the cookie consent, and have a GitHub repository configured, the consent needs to be given by the user. The GitHub cookie will be automatically added:

https://github.com/squidfunk/mkdocs-material/blob/13655d68f5c3514f1a03733ef7b2ecb96bc0d2fe/src/partials/consent.html#L30-L34

If this is not the case, please create an issue with a minimal reproduction that showcases the issue.

**Comment by white-gorilla** [comment]
Thank you very much @squidfunk for this very clear answer. It's OK with me.
